### PR TITLE
added template input plugin

### PR DESCRIPTION
**Pull request by l-d-j**
Implemented input template plugin. It can be run like: 
```
dock --input template --input-arg template_path=/tmp/template.json --input-arg replace=IMGNAME=jboss-docker-base,TESTS=node
```
then following json:
``` json
{
  "git_url": "http://git.app.eng.bos.redhat.com/git/jboss-dockerfiles.git",
  "git_commit": "master",
  "git_dockerfile_path": "standalone/eap/6.4",
  "image": "IMGNAME",
  "prepublish_plugins": [{
    "name": "test_built_image",
    "args": {
      "image_id": "BUILT_IMAGE_ID",
      "git_uri": "https://github.com/jboss-dockerfiles/tools",
      "git_commit": "master",
      "tests_git_path": "test/docker_test_base.py",
      "tests": "TESTS"
    }
  }]
}
```
will be translated to 
``` json
{
  "git_url": "http://git.app.eng.bos.redhat.com/git/jboss-dockerfiles.git",
  "git_commit": "master",
  "git_dockerfile_path": "standalone/eap/6.4",
  "image": "jboss-docker-base",
  "prepublish_plugins": [{
    "name": "test_built_image",
    "args": {
      "image_id": "BUILT_IMAGE_ID",
      "git_uri": "https://github.com/jboss-dockerfiles/tools",
      "git_commit": "master",
      "tests_git_path": "test/docker_test_base.py",
      "tests": "none"
    }
  }]
}
```

### CODE REVIEW

File: dock/plugins/input_template.py [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+            with open(self.template_path, 'r') as build_cfg_fd:
+                build_cfg_json = json.load(build_cfg_fd)
+        except ValueError:
+            self.log.error("couldn't decode json from file '%s'", self.template_path)
+            return None
+        except IOError:
+            self.log.error("couldn't read json from file '%s'", self.template_path)
+            return None
+        else:
+            for change in self.replace.split(','):
```
Comment (TomasTomecek) [inline dock/plugins/input_template.py]: what's the reasoning behind this?

**Comment by TomasTomecek** [comment]
This looks like a very good idea! I really like it.

File: dock/plugins/input_template.py (hunk 1 above)
Comment (l-d-j) [inline dock/plugins/input_template.py]: Its just to provide us possibility to have one json file (template) for all of our builds - our builds json differs only in image names and tags of which tests to run.
 The template its easily maintainable/extendable without changing code (I don't want to have config hard coded)

**Comment by TomasTomecek** [comment]
After a discussion on IRC, I've decided to resolve this more generally. Therefore one might be able to substitute values in build json from command line using option `--substitute`.

Thanks for your contribution!

https://github.com/DBuildService/dock/commit/15c6790b0aac14a254c404c689afdcf2597ed81c

**Pull request closed without merging**